### PR TITLE
Return team id when adding team

### PR DESCRIPTION
**Pull request by jonathan-dorsey**
This pull request updates the Team code to return the computed id value for newly created teams. This is necessary in support of adding team resource functionality in the terraform provider. See grafana/terraform-provider-grafana#120

Also updated the test to use and validate the newly returned id value. While there, modified the Team code to use a value instead of a pointer. This is more consistent with some of the similar code in org, alert, user...

NOTE: A better approach to these methods may be to accept an instance of Team that contains all the values for creating a new team object. Then a fully populated Team instance could be returned. This would work around the need to continually add parameters (and maybe return values for computed fields) as fields are added to the Team object in Grafana. Though I think this is likely a better long-term approach, I instead followed the patterns set out in other resources for consistency sake.

### CODE REVIEW

File: team.go [hunk 1]
```diff
@@ -79,18 +79,31 @@ func (c *Client) Team(id int64) (*Team, error) {
 // AddTeam makes a new team
 // email arg is an optional value.
 // If you don't want to set email, please set "" (empty string).
-func (c *Client) AddTeam(name string, email string) error {
+// When team creation is successful, returns the team id
```
Comment (aknuds1) [inline team.go]: I think it's best to standardize on ID as short for identity:
```suggestion
// When team creation is successful, returns the team ID.
```

File: team.go [hunk 2]
```diff
@@ -79,18 +79,31 @@ func (c *Client) Team(id int64) (*Team, error) {
...
-	return c.request("POST", path, nil, bytes.NewBuffer(data), nil)
+	tmp := struct {
+		Id int64 `json:"teamId"`
+	}{}
+
+	err = c.request("POST", path, nil, bytes.NewBuffer(data), &tmp)
+
+	if err != nil {
+		return id, err
+	}
```
Comment (aknuds1) [inline team.go]: It's preferable to co-locate the error check with the origin of the error, with readability in mind:
```suggestion
	err = c.request("POST", path, nil, bytes.NewBuffer(data), &tmp)
	if err != nil {
		return id, err
	}
```

File: teams_test.go [hunk 3]
```diff
@@ -158,10 +158,13 @@ func TestAddTeam(t *testing.T) {
 	name := "TestTeam"
 	email := ""
 
-	err := client.AddTeam(name, email)
+	id, err := client.AddTeam(name, email)
 	if err != nil {
 		t.Error(err)
 	}
+	if id == 0 {
+		t.Error("AddTeam returned an invalid id.")
```
Comment (aknuds1) [inline teams_test.go]: Error strings shouldn't be punctuated:
```suggestion
		t.Error("AddTeam returned an invalid ID")
```

File: team.go [hunk 4]
```diff
@@ -79,18 +79,31 @@ func (c *Client) Team(id int64) (*Team, error) {
...
 	}
 	data, err := json.Marshal(team)
 	if err != nil {
-		return err
+		return id, err
 	}
 
-	return c.request("POST", path, nil, bytes.NewBuffer(data), nil)
+	tmp := struct {
+		Id int64 `json:"teamId"`
```
Comment (aknuds1) [inline team.go]: [Words in names that are initialisms or acronyms (e.g. "URL" or "NATO") have a consistent case.](https://github.com/golang/go/wiki/CodeReviewComments#initialisms)
```suggestion
		ID int64 `json:"teamId"`
```

File: team.go [hunk 5]
```diff
@@ -79,18 +79,31 @@ func (c *Client) Team(id int64) (*Team, error) {
...
+	}{}
+
+	err = c.request("POST", path, nil, bytes.NewBuffer(data), &tmp)
+
+	if err != nil {
+		return id, err
+	}
+
+	return tmp.Id, err
```
Comment (aknuds1) [inline team.go]:
```suggestion
	return tmp.ID, err
```

**Review by aknuds1 — CHANGES_REQUESTED** [review]
Generally looks good to me, but spotted nits.

**Review by aknuds1 — APPROVED** [review]

**Comment by jonathan-dorsey** [comment]
@aknuds1 - thanks for the feedback. Suggestions incorporated. It's worth noting that most of the things you mentioned were done that way just for consistency sake with the org. I used that file as a template for the changes. We may want to enter a new issue to clean up the other files so the coding style is consistent everywhere in the new fork. Thanks again!

**Comment by aknuds1** [comment]
@jonathan-dorsey Gotcha, I think it would be best to enable style linting, but that requires more work. I suggest we do that through a dedicated PR.

**Comment by aknuds1** [comment]
I think @marefr wants to have a look, but will be busy until the end of the week.

**Comment by aknuds1** [comment]
Actually no, this is good to go.

**Comment by aknuds1** [comment]
Thanks!

**Pull request merged**